mozetl/engagement/churn/job.py

```diff
@@ -114,6 +114,33 @@ def clean_new_profile(new_profile):
     return new_profile.select(utils.build_col_expr(select_expr))
 
 
+def join_and_coalesce(lhs, rhs, on, col):
```

Comment:
I do think that this function now needlessly complicates the code. Up to you to leave it in.